src/utils/simple_fetch.js

```diff
@@ -48,7 +51,9 @@ function error (e) {
     return resolve({
```

Comment:
Why do we `resolve` a promise, when an error actually occurred? I feel like this behaviour is causing weird things later on, like having to actually look into the actual response to determine whether something went wrong in the first place.

I feel like that's the only reason why there's a `response.ok` attribute.

Reply:
I need one format (the response format) to fit the later promise pipeline, which decides whether to reject or resolve the `fetch` promise. I don't want the client to use different error formats for connection error and e.g. ResourceNotFound.